### PR TITLE
feat: add CreateWorkspaceUser and DropWorkspaceUser handlers

### PR DESCRIPTION
**Pull request by vojtabiberle**
## Summary
- Implement `CreateWorkspaceUserHandler` — creates a new service account with IAM roles and dataset OWNER access (no dataset creation)
- Implement `DropWorkspaceUserHandler` — revokes IAM, cancels jobs, deletes SA (no dataset deletion)
- Register both handlers in `HandlerFactory`
- Add `createWorkspaceCredentialsUserName()` to `NameGenerator` with `-qs-{rand4}` suffix

## Related PRs
- **connection**: https://github.com/keboola/connection/pull/6850
- **storage-backend (proto)**: https://github.com/keboola/storage-backend/pull/285

### CODE REVIEW

**Comment by romanbracinik** [comment]
👁️

File: tests/functional/UseCase/Bucket/ShareLinkBucketTest.php [hunk 1]
```diff
@@ -403,7 +403,13 @@ public function testShareUnshareLinkedBucket(): void
         // after unshare the table is not available
         // in connection you can't just unshare a bucket that is lined up first so this is an edge case
         // handled in connection
-        $this->expectException(BadRequestException::class);
-        $testTableBefore->exists();
+        try {
```
Comment (romanbracinik) [inline tests/functional/UseCase/Bucket/ShareLinkBucketTest.php:409]: toto mi pride divne ze sa to meni, pridavame nejake veci k WS, preco je to upravene robilo to nejake problemy?  Uplne sme takto vyhodili to co to testovalo

File: src/Handler/Workspace/CreateUser/CreateWorkspaceUserHandler.php [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+        // create service account
+        $iamService = $this->clientManager->getIamClient($credentials);
+        $projectName = 'projects/' . $projectCredentials['project_id'];
+        $newServiceAccountName = sprintf(
+            '%s/serviceAccounts/%s@%s.iam.gserviceaccount.com',
+            $projectName,
+            $newWsServiceAccId,
+            $projectCredentials['project_id'],
+        );
```
Comment (romanbracinik) [inline src/Handler/Workspace/CreateUser/CreateWorkspaceUserHandler.php]: nejak podobne asi vytvarme SA ked zakladame WS o by mohla byt nejaka spolocna metoda aby sme to nerobili na dvoch miestach.

File: tests/functional/UseCase/Workspace/CreateDropWorkspaceUserTest.php [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+use Keboola\TableBackendUtils\Escaping\Bigquery\BigqueryQuote;
+use Retry\BackOff\ExponentialBackOffPolicy;
+use Retry\Policy\CallableRetryPolicy;
+use Retry\RetryProxy;
+use Throwable;
+
+/**
+ * @group sync
+ */
+class CreateDropWorkspaceUserTest extends BaseCase
```
Comment (romanbracinik) [inline tests/functional/UseCase/Workspace/CreateDropWorkspaceUserTest.php:46]: Asi by som pridal testy, ze ten novy uzivatel, vidi read only veci, nemozem vytvorit bucket, nemoze zapisat do RO veci a pod. vlastne to co testujeme v \Keboola\StorageDriver\FunctionalTests\UseCase\Workspace\CreateDropWorkspaceTest kedze je to novy uzivatel tak aby bolo zabezpecene ze sa nam nejak nerozhodia permissions

**Review by romanbracinik — CHANGES_REQUESTED** [review]
Vylepsil by som tie testy

File: tests/functional/UseCase/Bucket/ShareLinkBucketTest.php (hunk 1 above)
Comment (vojtabiberle) [reply tests/functional/UseCase/Bucket/ShareLinkBucketTest.php:409]: This is not related to workspace user changes — it's a fix for a flaky test exposed by the dependency upgrades (PHPUnit 12, google/cloud-bigquery-analyticshub, google/gax).

The original code used `$this->expectException(BadRequestException::class)` but GCP can also throw `Google\ApiCore\ApiException` with status `INVALID_ARGUMENT` for the same scenario (accessing a table after the linked bucket is unshared). The CI was failing with:

```
Failed asserting that exception of type "Google\ApiCore\ApiException" matches expected exception
"Google\Cloud\Core\Exception\BadRequestException".
Message was: "Source dataset not found..."
```

The test still verifies the same behavior (that accessing the table after unshare throws an error), it just accepts both exception types that GCP may throw.

File: src/Handler/Workspace/CreateUser/CreateWorkspaceUserHandler.php (hunk 2 above)
Comment (vojtabiberle) [reply src/Handler/Workspace/CreateUser/CreateWorkspaceUserHandler.php]: jo, super nápad, udělal jsem

File: tests/functional/UseCase/Workspace/CreateDropWorkspaceUserTest.php (hunk 3 above)
Comment (vojtabiberle) [reply tests/functional/UseCase/Workspace/CreateDropWorkspaceUserTest.php:46]: pravda, testy přidány

**Review by romanbracinik — APPROVED** [review]
ok, dik za opravy 🙆‍♂️
